### PR TITLE
lib/fs: Sync does not fail (fixes #4432)

### LINKED ISSUE #4432: fsync on directories fails if directory is located in a samba share
Since samba does not support fsync on directories the syncthing daemon gives a lot of "invalid argument" error. It would be great if fsync could be disabled/ignored for directories. 
I think this behaviour is already the default case for windows (5ec1490), but I would like to use it on Linux.

Syncthing Version: v0.14.39
OS Version: Linux / Ubuntu 17.04

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
...

### CODE REVIEW

**Comment by calmh** [comment]
Well this is the lazy one leaving all kinds of `if err != nil { log and shit }` that are now dead code... If we don't care whether this works (which I'm fine with in the regular sync path) the clean solution is to explicitly ignore the error at the call site.

**Comment by AudriusButkevicius** [comment]
Fair point.

**Comment by AudriusButkevicius** [comment]
Updated

File: lib/config/folderconfiguration.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (f *FolderConfiguration) CreateMarker() error {
 		}
 		if dir, err := fs.Open("."); err == nil {
 			if serr := dir.Sync(); err != nil {
```
Comment (calmh) [inline lib/config/folderconfiguration.go]: There is a different bug here where the `if` sets `serr` bug checks `err`

Comment (calmh) [reply lib/config/folderconfiguration.go]: Eh, I can fix that

**Comment by calmh** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as 622b614f315b1ba145f81720e46d31ed7407c569. Thanks, @AudriusButkevicius!

**Pull request closed without merging**